common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1463,6 +1482,32 @@ def save_problem(self, data):
         event_info['answers'] = answers
         _ = self.runtime.service(self, "i18n").ugettext
 
+        # Anonymous users cannot save problem.
+        if not self.runtime.user_id:
+            event_info['failure'] = 'anonymous_user'
+            self.track_function_unmask('save_problem_fail', event_info)
+            next_url = urlquote_plus(reverse('jump_to', kwargs={'course_id':self.location.course_key, 'location':self.location}))
+
+            regiteration_link = u'{lms_root}/register?next={next_url}'.format(
```

Comment:
Can we use reverse('register_user') here?

Reply:
Done